parsl/dataflow/dflow.py

```diff
@@ -1199,15 +1202,16 @@ def cleanup(self) -> None:
             if not executor.bad_state_is_set:
                 if isinstance(executor, BlockProviderExecutor):
                     logger.info(f"Scaling in executor {executor.label}")
-                    job_ids = executor.provider.resources.keys()
-                    block_ids = executor.scale_in(len(job_ids))
-                    if self.monitoring and block_ids:
-                        new_status = {}
-                        for bid in block_ids:
-                            new_status[bid] = JobStatus(JobState.CANCELLED)
-                        msg = executor.create_monitoring_info(new_status)
-                        logger.debug("Sending message {} to hub from DFK".format(msg))
-                        self.monitoring.send(MessageType.BLOCK_INFO, msg)
+                    if executor.provider:
```

Comment:
i'm going to move this into a separate PR because it isn't directly task vine related. Parsl shutdown has been pretty fragile, so I'd like to keep this as an explicit commit in our history. You don't need to do anything to make that happen.